src/Prismic/SearchForm.php

```diff
@@ -222,99 +246,125 @@ public function submit()
      * This uses a copy of the SearchForm with a page size of 1 (the smallest
      * allowed) since all we care about is one of the returned non-result
      * fields.
-     *
-     * @return int Total number of results
-     *
-     * \throws RuntimeException
      */
-    public function count()
+    public function count() :? int
     {
-        return $this->pageSize(1)->submit_raw()->total_results_size;
+        $response = $this->pageSize(1)->submitRaw();
+        return isset($response->total_results_size)
+               ? (int) $response->total_results_size
+               : null;
     }
 
     /**
-     * Set the query's predicates themselves.
-     * You can pass a String representing a query as parameter, or one or multiple Predicates to build an "AND" query
-     *
-     * @return Prismic::SearchForm a clone of the SearchForm object with the new predicate or predicates added
+     * Set query predicates
+     * You can provide a single string, or one or multiple Predicate instances to build an "AND" query
      */
-    public function query()
+    public function query(...$params) : self
     {
-        $numargs = func_num_args();
-        if ($numargs == 0) return clone $this;
-        $first = func_get_arg(0);
-        if ($numargs == 1 && is_string($first)) {
-            return $this->set("q", $first);
+        // Filter empty args and return early if appropriate
+        $params = array_filter($params);
+        if (empty($params)) {
+            return clone $this;
         }
-        if ($numargs == 1 && is_array($first)) {
-            $predicates = $first;
-        } else {
-            $predicates = func_get_args();
+        $first = current($params);
+        // Unpack a single array argument
+        if (count($params) === 1 && is_array($first)) {
+            $params = $first;
+        }
+        $this->assertValidQueryParameters($params);
```

Comment:
should execute that before no? like in the top of `query` after array_filter I guess.

Reply:
yes maybe for the empty check for if it's an array with a single element, should we validate that it's actually a predicate first?